src/common/filehistorycmn.cpp

```diff
@@ -135,20 +136,43 @@ void wxFileHistoryBase::AddFileToHistory(const wxString& file)
     m_fileHistory.insert(m_fileHistory.begin(), file);
     numFiles++;
 
-    // update the labels in all menus
+    RefreshLabels();
+}
+
+void wxFileHistoryBase::RefreshLabels()
+{
+    size_t i;
+    size_t numFiles = m_fileHistory.size();
+
+    // If no files, then no need to refresh the menu
+    if ( numFiles == 0 )
+        return;
+
+    // Use the first file as the current path
+    wxFileName firstFn(m_fileHistory[0]);
```

Comment:
It looks like it could be simpler if we just did
```suggestion
    const wxString firstPath = wxFileName(m_fileHistory[0]).GetPath();
```
and then compared the path with it below.